File: a-honey/brute_force/소수찾기.py
# 가능한 조합을 전부 만들어서 소수인지 판단

from itertools import permutations

# 소수를 판단하는 함수
def is_prime(num):
    if num < 2:
        return False
    for i in range(2, int(num**.5) + 1): # num의 제곱근 올림까지만 판단
        if num % i == 0:
            return False
    return True

def solution(numbers):
    answer_num_array = set()
    
    for i in range(1, len(numbers) + 1):
        perms = permutations(numbers, i)
        for perm in perms:
            num = int(''.join(perm))
            if is_prime(num):
                answer_num_array.add(num)
        
        
    return len(answer_num_array)

File: a-honey/brute_force/소수찾기.js
function isPrime (num) {
  if (num < 2) return false;
  
  for (let i = 2; i <= Math.sqrt(num); i++) {
    if (num % i === 0) return false;
  }
  
  return true;
}

function getPermutations(nums_array) {
  if (nums_array.length === 1) return [nums_array];
  const nums = [];
  nums_array.forEach((fixed, index, original) => {
    const rest = [...original.slice(0, index), ...original.slice(index+1)];
    const permutations = getPermutations(rest);
    
    permutations.forEach((permutation) => {
      nums.push([fixed, ...permutation])
    })
  })
  return nums;
}

function solution(numbers) {
    const cases = getPermutations(numbers.split(''));
    const set_cases = new Set();
    let count = 0;
  
    cases.forEach((num_case) => {
      for (let i = 1; i <= num_case.length; i++) {
        const num = +(num_case.slice(0, i).join(''));
        set_cases.add(num)
      }
    })
  
    set_cases.forEach((num) => {
      if (isPrime(num)) count++;
    })
    return count;
}

console.log(isPrime(5))
console.log(solution("17"));
console.log(solution("011"));
